### DISCUSSION TITLE
Is it possible to use a DateRangePicker with react-hook-form ?

### DISCUSSION THREAD

**Discussion by ghost** [react-hook-form/react-hook-form#6793] [Q&A] [answered]
Hi everyone,

I have a question regarding an implementation of react-hook-form with an custom DateRangePicker from the react-nice-dates package.

For now, I coded this without a Controller : 
```
import { DateRangePicker, START_DATE, END_DATE } from "react-nice-dates";

[...]

<DateRangePicker
      startDate={startDate}
      endDate={endDate}
      onStartDateChange={setStartDate}
      onEndDateChange={setEndDate}
      minimumDate={Date.parse("01 Jan 1970 00:00:00 GMT")}
      minimumLength={1}
      format="dd/MM/yyyy">
      {({  startDateInputProps, endDateInputProps, focus }) => (
        <div className="date-range">
              <input
                // {...register("startDate")}
                className={`input${focus === START_DATE ? " -focused" : "" }`}
                {...startDateInputProps}
                placeholder="Start date"
              />
              <input
                // {...register("endDate")}
                className={`input${focus === END_DATE ? " -focused" : ""}`}
                {...endDateInputProps}
                placeholder="End Date"
              />
        </div>
      )}
    </DateRangePicker>
                      
```

As you can see, i've got two custom input inside one react component.

So how do you implement a Controller when the component is splited into two parts ?

Thank you for your help.

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api/usecontroller/controller

examples: https://codesandbox.io/s/react-hook-form-v7-controller-5h1q5

**Comment by ghost** [comment]
Thank you for your links, but i'm looking for an exemple for DateRangePicker with two custom inputs (startDate and endDate) at the same time.

Your sample demonstrates only a single ReactDatePicker.

Check it out here : [https://reactnicedates.hernansartorio.com/#overview](https://reactnicedates.hernansartorio.com/#overview )

**Comment by Moshyfawn** [comment, marked as answer]
Hi, @pickapackfr! Take a look at [this CSB](https://codesandbox.io/s/withered-firefly-ufogs?file=/src/Demo.tsx) for RHF use with [the `<DateRangePicker />`](https://reactnicedates.hernansartorio.com/#overview) from [react-nice-dates](https://github.com/hernansartorio/react-nice-dates).
_P.S. I've never used the lib, so there's some weirdness with it's input class styles and some issue with children, but I've used the docs example, so, should be fine. Other then that, the example should suffice._

**Comment by hdang09** [comment]
You can try with this: https://codesandbox.io/s/reacthookform-with-reactdatarange-rmcglg
